docs/design.md

```diff
@@ -131,3 +139,11 @@ This overall process of executing Upgrade Steps is illustrated below.
 To define a new custom procedure for performing a cluster upgrade, a developer should:
 - Create a new implementation of the `ClusterUpgrader` that defines a unique order of `UpgradeStep`s.
 - Implement any missing or new `UpgradeStep`s that need to be performed.  
+
+### Ready to upgrade criteria
```

Comment:
We are technically mixing criteria here. 
Ready to upgrade =  system time spec: upgradeAt 
where as 
UpgradeConfig validation = version is greater then current.

Reply:
I've revised this and split them out in commit 2dc8969